### ISSUE TITLE
Tutorial 09: Update to EmbeddingRetriever Training

### ISSUE THREAD

**Issue by bglearning** [deepset-ai/haystack-tutorials#35]
## Overview

With deepset-ai/haystack#2887, we replaced DPR with `EmbeddingRetriever` in Tutorial 06. 

Now, we might want to do the same for Tutorial 09 which covers training (or fine-tuning) a DPR Retriever model.

**Q1.** Should we go ahead with this switch? Any reason keeping DPR might be better?

Alternatively, we could create one for each. I guess depends on which we want to demonstrate plus what we think might be valuable for users.

## Training EmbeddingRetriever

Only the sentence-encoder variant of `EmbeddingRetriever` can be trained. 

Its `train` method does some data setup and then calls the `fit` method on `SentenceTransformer` (from the `sentence_transformers` package). 

Input data format is:
```
[
{”question”: …, “pos_doc”: …, “neg_doc”: …, “score”: …}, 
... 
]
```
It uses `MarginMSELoss` (as part of the GPL procedure). 

**Q2.** If we were to demonstrate its training, which data could be best to use? GPL et al. seem to use MSMARCO but then we need cross-encoder scores for the `score` above, right? So there doesn't seem to be a download-and-use form of dataset available?

RFC: @brandenchan @vblagoje @agnieszka-m (please loop in anyone else if necessary)
cc: @mkkuemmel

**Comment by vibha0411** [comment]
This is a related question @bglearning 
I see that
[https://colab.research.google.com/drive/1Tz9GSzre7JfvXDDKe7sCnO0FMuDViMnN?usp=sharing#scrollTo=TD2PZuuNTpQ3]
PseudoLabelGenerator runs basically
**Question generation - optional step
Negative mining
Pseudo labeling (margin scoring)**
steps.
But in a case where i already have the positive and negative documents for a question, I would not be requiring the negative mining step, however as per my understanding I still require the margin scoring to perform the training.

So how do I go about this scenario?
Kindly help.

**Comment by bglearning** [comment]
Hey @vibha0411 ,

If you use MultipleNegativesRankingLoss (`train_loss='mnrl'`, currently the default), the scores aren't required. [1]

In fact, for MNRL even having the negative docs is optional because it considers all other docs in a batch as negative. The [MNRL example colab notebook](https://colab.research.google.com/drive/1cZxlQN8yl57Yuq77aHlX9YyfJAofVfzl?usp=sharing) linked above might be useful. It only uses positive docs but of course it's good if you already have negatives and you can pass those too. 

Roughly would be something like:

```python
embedding_retriever = EmbeddingRetriever(...)
training_data = [
    {"question": ..., "pos_doc": ..., "neg_doc": ...}, 
    ...
]
embedding_retriever.train(training_data=training_data, train_loss='mnrl')  
```

[1] Margin scoring would be required when training with `margin_mse`.

**Comment by vibha0411** [comment]
Thank you for your reply.
Yes, I did have a look at MNRL as well.
But in my case i since the customer feedback contains the explicit negative feedback, I am looking for a loss function that can leverage this handpicked negative feedback.

Hence I thought it might be more sensible to go with margin scoring.. 

But not sure what is the best way to go about it.
